### DISCUSSION TITLE
Please help us define materials to be delivered

### DISCUSSION THREAD

**Discussion by heathermeeker** [Open-Weights/Definition#6] [Announcements]
Thanks to everyone for the PRs so far!

I would particularly like input on the item in the definition that described what needs to be delivered, in order for a model to be open. In other words, what is the analog of source code, for open source software? 

Also, is there a format in which it should be delivered?

**Comment by AzulGarza** [comment]
Thank you for opening this interesting discussion!

I think when we talk about making a machine learning model open-source, the analogy to the source code in software is the full pipeline of generating the model, along with the model weights. Here's what I believe should be included:

**1. Training Code and Environment:**
The source code used for training the models should be shared, detailing:

   - The programming language used, along with its specific version.
   - All open-source libraries utilized, specifying their respective versions to ensure reproducibility.
   - The operating system and hardware on which the model was trained, providing a rough estimate of the expected training time.
   - A Docker image can be especially beneficial to encapsulate the development environment and eliminate dependency conflicts.

**2. Model Weights and Usage Instructions:**
For the neural network weights (NNW), the following should be provided:

   - A publicly accessible URL where the model weights are hosted.
   - A comprehensive guide on how to use the pre-trained models with the aforementioned requirements. This should include sample code illustrating how to load the model weights and make predictions.

**3. Repository for Information:**
All this information should ideally be stored and organized in a GitHub/GitLab repository, with proper documentation to assist contributors.

**Comment by mergenthaler** [comment]
I think these people could contribute to this discussion: @julien-c, @williamFalcon, @jph00, @rasbt, @clmnt, @dongreenberg.

**Comment by rasbt** [comment]
> The source code used for training the models should be shared, detailing:

In particular, I'd say the definition is that source code should be made **freely available and may be redistributed and modified by anyone**. I.e., not only be shared, but shared freely. And modifiable to one's liking.

> making a machine learning model open-source

if we are talking about making "the model" open source, then the weights should also be made freely available etc.

EDIT:

Is it always feasible? Probably not. But in this case it then shouldn't be called open source. I honestly don't blame people if they can't make everything open-source. It's more false advertising if some people/companies put an open-source label on something that isn't. It's like saying "free unlimited internet" but then having a "500 Gb / month" data volume cap.
